src/smartdrive/smartdrive.cxx

```diff
@@ -179,11 +182,11 @@ SmartDrive::Run_Seconds(int motor_id, int direction, uint8_t speed, uint8_t dura
             ctrl |= SmartDrive_CONTROL_GO;
         if ( direction != SmartDrive_Dir_Forward )
             speed = speed * -1;
-        if ( motor_id != SmartDrive_Motor_ID_2) {
+        if ((motor_id & SmartDrive_Motor_ID_2)!= 0) {
```

Comment:
These two if statement adjustments appear to be the only functional changes.   Are the other added lines and changes to spacing needed?